stingray/powerspectrum.py

```diff
@@ -625,6 +695,10 @@ class AveragedPowerspectrum(AveragedCrossspectrum, Powerspectrum):
     nphots: float
         The total number of photons in the light curve
 
+    legacy: bool
+        Use the legacy machinery of AveragedPowerspectrum. This might be useful to compare
+        with old results, and is also needed to use light curve lists as an input, to
+        conserve the spectra of each segment, or to use the large_data option.
```

Comment:
So longer-term, does this mean we will always need to have two interfaces? One that's the new one, and for large data sets, the legacy one? Or are you planning to transition the `large_data` option to some version of the new interface, too?

Reply:
I think we are going into a direction where the large_data option has no practical advantage, because the new machinery can be adapted to handle large light curves as well. We'll see if there is a next level of optimization where the large_data option is still useful. What do you think?